flake.nix

```diff
@@ -7,11 +7,53 @@
 
   outputs = { self, nixpkgs }: {
 
-    lib = {
-      getHashSubdomain = nixosCfg :
-        let prefixLength = nixpkgs.lib.stringLength "/nix/store/";
-            hash = builtins.substring prefixLength 32 nixosCfg.config.system.build.toplevel.drvPath;
-        in hash + ".hash.garnix.me";
+    nixosModules.garnix = { lib, config, ... }:
+      let cfg = config.garnix.persistence;
+      in {
+
+        options.garnix.persistence = {
+          enable = lib.mkEnableOption "Enable persistence in garnix deploys";
+
+          name = lib.mkOption {
+            type = lib.types.str;
+            description = "A unique name to identify this persistence.";
+          };
+        };
+
+        config = lib.mkIf cfg.enable {
+          services.openssh.enable = true;
+
+          security.sudo = {
+            enable = true;
+            execWheelOnly = true;
+            wheelNeedsPassword = false;
+          };
+
+          nix.settings.trusted-users = [ "garnix" ];
+
+          users.users.garnix = {
+            description = "A user garnix uses for redeploying ${cfg.name}";
+            isNormalUser = true;
+            createHome = false;
+            openssh.authorizedKeys.keys = [
+              "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIJTPguYAuqp6qCU43u8g2hgWz4MLCEPPyoVPYO53qB+t garnixServer@garnix.io"
+            ];
+            extraGroups = [ "wheel" ];
+          };
+        };
+      };
```

Comment:
Should we use `lib.mkForce` for all of these options? Otherwise users can just overwrite the settings and break persistence.

Reply:
AFAIK, the way this works is:

`mkForce` would hide the fact we're forcing, no?
If the user sets these settings to the same values as us, nothing happens.
If the user sets these settings to a different value, they get an error.

So if that is true, then I think it's better to leave them as is?

Reply:
> AFAIK, the way this works is:
> 
> `mkForce` would hide the fact we're forcing, no?

Do you mean that we would silently ignore any user settings that happen at a lower priority then what `mkForce` has? Then yes. (And I do think this is not ideal, I'd prefer they get an error.)

> If the user sets these settings to the same values as us, nothing happens.
> If the user sets these settings to a different value, they get an error.

Not sure I understand. Do you mean that if we *don't* use `mkForce` and the user sets the values (with the same priority) to the same values as ours, it'll just work? And if they set it to something different, they'll get a helpful error?

What I was trying to prevent is the following situation: Someone inadvertently (e.g. by using someone else's module) sets these options at a higher priority than what we have in this PR right now. And those settings will then silently overwrite ours and break persistence.

Maybe we should not use `mkForce` but instead implement a module that

1. sets these config options to the right values, *and*
2. asserts that they aren't overwritten, giving a nice error message otherwise.